1.preprocessing-data/ICutils.py

```diff
@@ -41,12 +43,17 @@ def load_pybasic_data(
 
     # This for loop will run through the specified directory, find images for a specific channel (channel name is in the file name metadata),
     # save paths to a list, and then save the names to a list after stripping the file_extension
-    for image_path in channel_path.iterdir():
-        image_path = str(image_path)
-        if channel in image_path:
-            image_files.append(image_path)
-            if image_path.endswith(file_extension):
-                image_names.append(image_path.strip(file_extension))
+    for plate_path in data_path.iterdir():
+        # print(plate_path)
+        for image_path in plate_path.iterdir():
+            image_path = str(image_path)
+            # print(image_path)
+            if plate in plate_path.name:
+                # print(plate)
+                if channel in image_path:
+                    image_files.append(image_path)
```

Comment:
@gwaybio 

I am currently not too concerned, but I do not know the consequences of running all the images at one.

I could code for the IC to happen per well for each plate, which would reduce the number of images run (might make it faster, not sure). 

But for CellProfiler, that would mean having a .sqlite file per well for each plate (based on how my current code works). I am not sure how that would impact using Cytosnake, because based on the documentation, the .sqlite files are per plate.

I think that CellProfiler can handle a lot of images, as long as it per plate. I would need to test this, like on the cell-health data, to see if I am right on this.

Reply:
> I could code for the IC to happen per well for each plate, which would reduce the number of images run (might make it faster, not sure).

That is an interesting suggestion. Have you observed consistent and substantially different illumination errors between wells?

Reply:
I would say I have noticed it between channels more than I have the wells. This is the only way that I can see having less images run.